### PR TITLE
Feature/remove dist folder and add release script

### PR DESCRIPTION
**Pull request by zenflow**
This PR extends #840 with the release script proposed there.

You can test this out with the `--dry-run` flag (i.e. run `npm run release -- --dry-run`), which has two effects:

1. You will not be prevented from running the script on a branch other than `master`, i.e. this branch.
2. The script won't do anything to interact with the "outside world", i.e. publishing to npm and pushing to Github will be skipped. This way you can run the script, verify the `dist` branch is as it should be, and then revert the changes with `git reset --hard HEAD~${numberOfCommits}` on each branch.

So please try creating a `dist` branch (should be branched from the branch you are "releasing"; for testing it will be this branch), adding a commit to this branch (branch you are "releasing", not the dist branch) bumping the version, and running `npm run release --dry-run` .

@limonte Are there any other flags or parameters needed here?..

 https://github.com/sweetalert2/sweetalert2/blob/a7b9fe868f7b2ad284c05e6c61a42905ceaa83c7/release.js#L45-L46

### CODE REVIEW

**Comment by zenflow** [comment]
Oh, when you're reverting the changes made by the script, you also have to remove the tag it created.

**Comment by limonte** [comment]
Rebased to master w/o any changes to the code itself.

> @limonte Are there any other flags or parameters needed here?
> await execute('npm publish')

Nope, as long as you are logged in with the account which have a permission to publish, `npm publish` will work w/o any additional flags/params.

File: utils/get-git-status.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const assert = require('assert')
+
+async function getGitStatus () {
+  const lines = (await execute('git status')).stdout.split(/\r?\n/)
+
+  const match = lines[0].match(/^On branch (\S+)$/)
+  assert.ok(match, 'Unable to determine current branch')
+  const currentBranch = match[1]
+
+  const cleanWorkingTree = lines.includes('nothing to commit, working tree clean')
```
Comment (limonte) [inline utils/get-git-status.js]: Is there any better way to check if working directory is clean? Depending on wording from the output seems very unstable. E.g. for my env it's working **directory**, not working **tree**:

Comment (zenflow) [reply utils/get-git-status.js]: Should probably be using something like this https://github.com/nodegit/nodegit

Comment (zenflow) [reply utils/get-git-status.js]: Though i do like the approach of simply executing shell commands.. They are easily repeatable manually. Tt's good to know the script is doing the *exact* same thing we are doing.

What if we just checked for "nothing to commit,"?

Comment (limonte) [reply utils/get-git-status.js]: > What if we just checked for "nothing to commit,"?

Fine by me :+1:

Comment (zenflow) [reply utils/get-git-status.js]: @limonte Done in 5cf385dc60655f36f29bb4e02e5aa4adba6b58bb

File: utils/get-git-status.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const assert = require('assert')
+
+async function getGitStatus () {
+  const lines = (await execute('git status')).stdout.split(/\r?\n/)
+
+  const match = lines[0].match(/^On branch (\S+)$/)
+  assert.ok(match, 'Unable to determine current branch')
+  const currentBranch = match[1]
+
+  const cleanWorkingTree = lines.find(line => line.startsWith('nothing to commit,'))
```
Comment (limonte) [inline utils/get-git-status.js]: The result should be converted to boolean, e.g.

```js
const cleanWorkingTree = !!lines.find(line => line.startsWith('nothing to commit,'))
```

Comment (zenflow) [reply utils/get-git-status.js]: 4e8d5482e9f743542e59812f9089db144eb83fb4

**Pull request merged**